rll/rightscale-mirrors.sh

```diff
@@ -0,0 +1,300 @@
+
+FREEZE_DATE=${FREEZE_DATE//-/}
+if [[ -z "$FREEZE_DATE" ]]; then
+  echo "Since FREEZE_DATE is not set, not configuring RightScale software mirrors"
+elif [[ ! "$FREEZE_DATE" =~ ^[0-9]{8}$ ]] && [[ ! "$FREEZE_DATE" = "latest" ]]; then
+  echo "If FREEZE_DATE is set, it must be either 'latest' or in the format YYYY-MM-DD"
+  exit 1
+fi
+today=$(date +"%Y%m%d")
+if [[ "$FREEZE_DATE" =~ ^[0-9]{8}$ ]] && [[ $FREEZE_DATE -gt $today ]]; then
+  echo "FREEZE_DATE can't be from a date in the future"
+  exit 1
+fi
+
+RUBYGEMS_FREEZE_DATE=${RUBYGEMS_FREEZE_DATE//-/}
+if [[ -z "$RUBYGEMS_FREEZE_DATE" ]]; then
+  echo "Since RUBYGEMS_FREEZE_DATE is not set, not configuring Rubygems mirror"
+elif [[ ! "$RUBYGEMS_FREEZE_DATE" =~ ^[0-9]{8}$ ]] && [[ ! "$RUBYGEMS_FREEZE_DATE" = "latest" ]]; then
+  echo "If RUBYGEMS_FREEZE_DATE is set, it must be either 'latest' or in the format YYYY-MM-DD"
+  exit 1
+fi
+today=$(date +"%Y%m%d")
+if [[ "$RUBYGEMS_FREEZE_DATE" =~ ^[0-9]{8}$ ]] && [[ $RUBYGEMS_FREEZE_DATE -gt $today ]]; then
+  echo "RUBYGEMS_FREEZE_DATE can't be from a date in the future"
+  exit 1
+fi
+
+if [[ -n "$FREEZE_DATE" ]] || [[ -n "$RUBYGEMS_FREEZE_DATE" ]]; then
+  if [[ -z "$MIRROR_HOST" ]]; then
+    echo "MIRROR_HOST cannot be blank, possible values are cf-mirror.rightscale.com or ENV:RS_ISLAND"
+  elif [[ ! "$MIRROR_HOST" =~ island|mirror ]]; then
+    echo "MIRROR_HOST (= $MIRROR_HOST) appears to be invalid, possible values are cf-mirror.rightscale.com or ENV:RS_ISLAND"
+    exit 1
+  fi
+fi
+
+distro=unknown
+distro_ver=unknown
+if which apt-get >/dev/null 2>&1; then
+  if [[ -f /etc/lsb-release ]]; then
+    . /etc/lsb-release
+    distro=${DISTRIB_ID:-unknown}
+    distro_ver=${DISTRIB_RELEASE:-unknown}
+    distro_codename=${DISTRIB_CODENAME:-unknown}
+  fi
+elif [[ -f /etc/centos-release ]]; then
+  if [[ $(cat /etc/centos-release) =~ ^([^0-9]+)\ ([0-9])\. ]]; then
+    distro="${BASH_REMATCH[1]}"
+    distro=${distro/Red Hat/RedHat}
+    distro_ver="${BASH_REMATCH[2]}"
+    distro_ver=$(echo $distro_ver | cut -d. -f1)
+  fi
+elif [[ -f /etc/redhat-release ]]; then
+  if [[ $(cat /etc/redhat-release) =~ ^([^0-9]+)\ ([0-9])\. ]]; then
+    distro="${BASH_REMATCH[1]}"
+    distro=${distro/Red Hat/RedHat}
+    distro_ver="${BASH_REMATCH[2]}"
+    distro_ver=$(echo $distro_ver | cut -d. -f1)
+  fi
+fi
+distro=$(echo $distro | cut -d' ' -f1 | tr [:upper:] [:lower:])
+arch_info=$(echo $(uname -i 2>/dev/null)$(uname -m 2>/dev/null))
+if [[ arch_info =~ 386|686 ]]; then
+  arch=i386
+else
+  arch=x86_64
+fi
+
+freeze_date_msg=$FREEZE_DATE
+if [[ -z "$FREEZE_DATE" ]]; then
+  freeze_date_msg='(unconfigured)'
+fi
+rubygems_freeze_date_msg=$RUBYGEMS_FREEZE_DATE
+if [[ -z "$RUBYGEMS_FREEZE_DATE" ]]; then
+  rubygems_freeze_date_msg='(unconfigured)'
+fi
+echo "Distro: $distro"
+echo "Distro version: $distro_ver"
+echo "Architecture: $arch"
+echo "OS Repositories Freeze Date: $freeze_date_msg"
+echo "Rubygems Freeze Date: $RUBYGEMS_FREEZE_DATE"
```

Comment:
So we do fail at some point below -- if the value of distro isn't centos/redhat/ubuntu then we error out in a case statement.